Automattic/wp-calypso
# Jetpack Agency Dashboard: Wait to load partner oAuth token before requesting sites

#### Proposed Changes

Currently the first request to `/jetpack-agency/sites` is failing with a 403 not authorized due to the auth not being included in the header. Subsequent requests succeed. This PR makes all the requests to `/jetpack-agency/sites` include the auth token.

This was occurring because requests to the partner API use a partner token instead of the regular user token. This partner token has to be retrieved from `/jetpack-licensing/partner` using the regular user token, and it is then set on the `wpcomJetpackLicensing` object which is used for all requests to the partner API.

The request to `/jetpack-agency/sites` was being made before the partner token was set on `wpcomJetpackLicensing`, so the request was failing to authenticate. Subsequent requests happened to be made and succeed because React happened to render after the token was set, but this was a race condition and it is possible that other behavior could have occurred.

In order to make the `/jetpack-agency/sites` request wait I have added a new `isPartnerOAuthTokenLoaded` boolean to the partner store which is set to `true` after the token is loaded. I then made the query dependent on this boolean according to [the dependent queries API](https://react-query.tanstack.com/guides/dependent-queries). Thus redux will cause the component which runs that query to re-render after the token is loaded and the sites will be retrieved.

The downside to this pattern is that every query to the partner API must remember to be dependent on `isPartnerOAuthTokenLoaded`, so I am open to other ideas which avoid this.

#### Testing Instructions



1. Open the network tab in your browser's dev tools and visit `jetpack.cloud.localhost:3001/dashboard`.
2. Verify that no requests to `/jetpack-agency/sites` fail.



Related to 1202076982646589-as-1202395639731514
